src/update_utils.c

```diff
@@ -0,0 +1,49 @@
+#include <errno.h>
+#include <gio/gio.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "update_utils.h"
+#include "context.h"
+
+gboolean r_copy_stream_with_progress(GInputStream *in_stream, GOutputStream *out_stream,
```

Comment:
I would have preferred not to mix the rename/move with changing the function's return value, but since it is not too invasive, it might still be ok.

Reply:
OK, I'll split it up.